lib/doWhilst.js

```diff
@@ -19,7 +20,7 @@ import onlyOnce from './internal/onlyOnce';
  * passes. The function is passed a `callback(err)`, which must be called once
  * it has completed with an optional `err` argument. Invoked with (callback).
  * @param {Function} test - synchronous truth test to perform after each
```

Comment:
I was thinking about the best way to document this.  For sure, add something to the `intro.md`, but we have lots of repeated docs where we describe the iteratee function as a node-style async function.  It would be nice to centralize the description of what Async accepts, with each Async method just describing the non-callback parameters.